networkx/drawing/layout.py

```diff
@@ -1233,6 +1235,225 @@ def arf_layout(
     return dict(zip(G.nodes(), p))
 
 
+def estimate_factor(n, swing, traction, speed, speed_efficiency, jitter_tolerance):
+    """
+    ForceAtlas2 helper function
+    Computes scaling factor for force
+    """
+    import numpy as np
+
+    # estimate jitter
+    opt_jitter = 0.05 * np.sqrt(n)
+    min_jitter = np.sqrt(opt_jitter)
+    max_jitter = 10
+    min_speed_efficiency = 0.05
+
+    other = min(max_jitter, opt_jitter * traction / n**2)
+    jitter = jitter_tolerance * max(min_jitter, other)
+
+    if swing / traction > 2.0:
+        if speed_efficiency > min_speed_efficiency:
+            speed_efficiency *= 0.5
+        jitter = max(jitter, jitter_tolerance)
+    if swing == 0:
+        target_speed = np.inf
+    else:
+        target_speed = jitter * speed_efficiency * traction / swing
+
+    if swing > jitter * traction:
+        if speed_efficiency > min_speed_efficiency:
+            speed_efficiency *= 0.7
+    elif speed < 1000:
+        speed_efficiency *= 1.3
+
+    max_rise = 0.5
+    speed = speed + min(target_speed - speed, max_rise * speed)
+    return speed, speed_efficiency
+
+
+def forceatlas2_layout(
+    G,
+    pos=None,
+    *,
+    n_iter=100,
+    jitter_tolerance=1.0,
+    scaling_ratio=2.0,
+    gravity=1.0,
+    distributed_action=False,
+    strong_gravity=False,
+    node_mass=None,
+    node_size=None,
+    weight=None,
+    dissuade_hubs=False,
+    linlog=False,
+    dim=2,
+):
+    """Forceatlas2 layout for networkx
+
+    See [1] for more info on the parameters
+
+    [1]: https://journals.plos.org/plosone/article/file?id=10.1371/journal.pone.0098679&type=printable
+    Parameters
+    ----------
+    G : nx.Graph
+       Netwrorkx graph
+    pos: dict or None
+       Optional starting positions
+    n_iter: int
```

Comment:
Maybe `max_iter` would be a better name since there is a stopping criteria that can result in exiting the loop early.

Reply:
Sure let's do that